java/java.editor/src/org/netbeans/modules/editor/java/GoToSupport.java

```diff
@@ -1193,47 +1204,60 @@ private void dumpArguments(List<? extends VariableElement> list, List<? extends
 
                 addSpace = true;
             }
-                
+
             result.append(')');
         }
 
         private void dumpThrows(List<? extends TypeMirror> list) {
             if (list.isEmpty()) {
                 return ;
             }
-            
+
             boolean addSpace = false;
-            
+
             result.append(" throws ");
-            
+
             for (TypeMirror t : list) {
                 if (addSpace) {
                     result.append(", ");
                 }
-                
+
                 result.append(getTypeName(info, t, true));
-                
+
                 addSpace = true;
             }
         }
-            
+
     }
-    
+
     private static String getTypeName(CompilationInfo info, TypeMirror t, boolean fqn) {
         return translate(Utilities.getTypeName(info, t, fqn).toString());
     }
-    
+
     private static String[] c = new String[] {"&", "<", ">", "\n", "\""}; // NOI18N
     private static String[] tags = new String[] {"&amp;", "&lt;", "&gt;", "<br>", "&quot;"}; // NOI18N
-    
+
     private static String translate(String input) {
         for (int cntr = 0; cntr < c.length; cntr++) {
             input = input.replace(c[cntr], tags[cntr]);
         }
-        
+
         return input;
     }
-    
+
+    static String findFileName(Element resolved) {
```

Comment:
What's the right way to do this without reflection, @jlahoda?

Reply:
I suspect the best way would be to add an API method somewhere into java.source.base, returning a "source file name" for Elements that originate in a classfile - presumably, it could suggest it returns the value of the attribute. Or, vaguely speak about some "source file name", which could work for all Elements, whether form class or from source.

`ElementUtilities` would normally be a fine place for a well-defined method, but I suspect this method may be a bit vague (and possibly single-purpose), so `SourceUtils` would sound better for a method like this.

Reply:
[e7d5437](https://github.com/apache/netbeans/pull/5152/commits/e7d543795366b9fc817b50973a698aa2e7ba4b26) introduces `SourceUtils.findSourceFileName` method.